src/classes/CRLP_ApiService_TEST.cls

```diff
@@ -0,0 +1,185 @@
+        System.assertEquals(2, rollups[0].rollupsMdt.size(), 'There should be two Rollup__mdt records associated with the CRLP_Rollup instance');
+
+        rollups = crlpApiSvc.getRollupDefsForRollupType(CRLP_RollupProcessingOptions.RollupType.ContactHardCredit.name());
+        System.assertEquals(1, rollups.size(), 'The collection of returned rollups should contain exactly 1 instance for ContactHardCredit');
+        System.assertEquals(2, rollups[0].rollupsMdt.size(), 'There should be two Rollup__mdt records associated with the CRLP_Rollup instance');
+
+        rollups = crlpApiSvc.getRollupDefsForRollupType(CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit.name());
+        System.assertEquals(1, rollups.size(), 'The collection of returned rollups should contain exactly 1 instance for AccountSoftCredit');
+        System.assertEquals(2, rollups[0].rollupsMdt.size(), 'There should be two Rollup__mdt records associated with the CRLP_Rollup instance');
+
+        rollups = crlpApiSvc.getRollupDefsForRollupType(CRLP_RollupProcessingOptions.RollupType.AccountContactSoftCredit.name());
+        System.assertEquals(1, rollups.size(), 'The collection of returned rollups should contain exactly 1 instance for AccountContactSoftCredit');
+        System.assertEquals(2, rollups[0].rollupsMdt.size(), 'There should be two Rollup__mdt records associated with the CRLP_Rollup instance');
+
+        rollups = crlpApiSvc.getRollupDefsForRollupType(CRLP_RollupProcessingOptions.RollupType.ContactSoftCredit.name());
+        System.assertEquals(1, rollups.size(), 'The collection of returned rollups should contain exactly 1 instance for ContactSoftCredit');
+        System.assertEquals(2, rollups[0].rollupsMdt.size(), 'There should be two Rollup__mdt records associated with the CRLP_Rollup instance');
+
+        rollups = crlpApiSvc.getRollupDefsForRollupType(CRLP_RollupProcessingOptions.RollupType.RecurringDonations.name());
+        System.assertEquals(1, rollups.size(), 'The collection of returned rollups should contain exactly 1 instance for RecurringDonations');
+        System.assertEquals(2, rollups[0].rollupsMdt.size(), 'There should be two Rollup__mdt records associated with the CRLP_Rollup instance');
+
+        rollups = crlpApiSvc.getRollupDefsForRollupType(CRLP_RollupProcessingOptions.RollupType.GAU.name());
+        System.assertEquals(1, rollups.size(), 'The collection of returned rollups should contain exactly 1 instance for GAU');
+        System.assertEquals(2, rollups[0].rollupsMdt.size(), 'There should be two Rollup__mdt records associated with the CRLP_Rollup instance');
+    }
+
+    /**
+     * @description Validate the getRollupDefsForRollupType returns an exception if the RollupType is null or invalid
+     */
+    @IsTest
+    static void validateGetRollupDefsForRollupTypeInvalidParameter() {
+        CRLP_ApiService crlpApiSvc = new CRLP_ApiService();
+
+        Test.startTest();
+        try {
```

Comment:
Since the same conditions are tested for various params, we can apply the same pattern to reduce number of statements. Also the verified error message should check for the param name (that probably should be a `CRLP_ApiService` `@TestVisible` constant).
```
for (String param : new String[] { null, 'NonExistentType' }) {
    Exception actualException = null;
    try {
       crlpApiSvc.getRollupDefsForRollupType(param);
    } catch (Exception e) {
        actualException = e;
    }
    System.assertNotEquals(null, actualException, 'An exception should be thrown for invalid param: ' + param);
    System.assert(actualException.getMessage().containsIgnoreCase('RollupType'), 'Message should contain "RollupType" param: ' + actualException.getMessage());
}
```

Reply:
Great idea!